Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -70,28 +70,35 @@ public class CustomerIOImplementation: CustomerIOInstance {
         body: RequestBody,
         jsonEncoder: JSONEncoder? = nil
     ) {
-        let isChangingIdentifiedProfile = profileStore.identifier != nil && profileStore.identifier == identifier
-        if isChangingIdentifiedProfile {
-            // TODO: add to background queue delete device token from currently registered profile.
+        let currentlyIdentifiedProfileIdentifier = profileStore.identifier
+        let isChangingIdentifiedProfile = currentlyIdentifiedProfileIdentifier != nil &&
+            currentlyIdentifiedProfileIdentifier != identifier
+
+        if let currentlyIdentifiedProfileIdentifier = currentlyIdentifiedProfileIdentifier,
```

Comment:
this some sort of garbage collection thing? why are we re-`let`ing the same var here?

Reply:
It's a common swift pattern to safely check a variable for null. 

Since `currentlyIdentifiedProfileIdentifier` is `String?`, if I ever want to use it I must do `currentlyIdentifiedProfileIdentifier!` which is dangerous. Instead, we can safely check by using `if let ...`

```swift
if let id = currentlyIdentifiedProfileIdentifier {
```

but, I don't like to introduce 2 variable names for the same 1 variable. That's why we use the same name:

```swift
if let currentlyIdentifiedProfileIdentifier = currentlyIdentifiedProfileIdentifier {
  // now, when you use `currentlyIdentifiedProfileIdentifier` in this scope, it's String not String?
```